src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -0,0 +1,169 @@
+
+from Utils.CertTools import cert, ckey
+from Utils.Timers import timeFunction
+from WMCore.Services.Rucio.Rucio import Rucio
+from WMCore.Services.pycurl_manager import RequestHandler
+from WMCore.WMException import WMException
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from WMCore.WorkerThreads.BaseWorkerThread import BaseWorkerThread
+from WMCore.DAOFactory import DAOFactory
+
+
+class WorkflowUpdaterException(WMException):
+    """
+    Specific WorkflowUpdaterPoller exception handling.
+    """
+
+
+class WorkflowUpdaterPoller(BaseWorkerThread):
+    """
+    Poller that does the actual work for updating workflows.
+    """
+
+    def __init__(self, config):
+        """
+        Initialize WorkflowUpdaterPoller object
+        :param config: a Configuration object with the component configuration
+        """
+        super(BaseWorkerThread, self).__init__()
+
+        myThread = threading.currentThread()
+        self.daoFactory = DAOFactory(package="WMCore.WMBS",
+                                     logger=myThread.logger,
+                                     dbinterface=myThread.dbi)
+        self.listActiveWflows = self.daoFactory(classname="Workflow.GetUnfinishedWorkflows")
+
+        # parse mandatory attributes from the configuration
+        self.config = config
+        self.rucioAcct = getattr(config.WorkflowUpdater, "rucioAccount")
+        self.rucioUrl = getattr(config.WorkflowUpdater, "rucioUrl")
+        self.rucioAuthUrl = getattr(config.WorkflowUpdater, "rucioAuthUrl")
+        self.rucioCustomScope = getattr(config.WorkflowUpdater, "rucioCustomScope",
+                                        "group.wmcore")
+        self.msPileupUrl = getattr(config.WorkflowUpdater, "msPileupUrl")
+
+        self.userCert = cert()
+        self.userKey = ckey()
+        self.rucio = Rucio(acct=self.rucioAcct,
+                           hostUrl=self.rucioUrl,
+                           authUrl=self.rucioAuthUrl)
+                           # configDict={'logger': self.logger})
+
+    @timeFunction
+    def algorithm(self, parameters=None):
+        """
+        Executed in every polling cycle. The actual logic of the component.
+        :param parameters: not really used. But keeping same signature as
+            the one defined in the super class.
+        :return: only what is returned by the decorator
+        """
+        logging.info("Running Workflow updater injector poller algorithm...")
+        try:
+            # retrieve list of workflows with unfinished Production
+            # or Processing subscriptions
+            wflowSpecs = self.listActiveWflows.execute()
+            if not wflowSpecs:
+                logging.info("Agent has no active workflows at the moment")
+                return
+
+            # figure out workflows that have pileup
+            puWflows = self.findWflowsWithPileup(wflowSpecs)
+
+            # otherwise, move on retrieving pileups
+            pileupMap = self.getPileupDocs()
```

Comment:
If `pileupMap` is not used in return there is no need to make this local variable, you should either change it to `_` or not assign anything at all from right hand side of this expression.

Reply:
As we work in the other tickets relevant to this component, it will be used.